src/Form/Type/ProductOptionValueFilter.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Lakion\SyliusElasticSearchBundle\Form\Type;
+
+use Doctrine\ORM\EntityRepository;
+use FOS\ElasticaBundle\Manager\RepositoryManagerInterface;
+use FOS\ElasticaBundle\Repository;
+use Lakion\SyliusElasticSearchBundle\Form\DataTransformer\ProductOptionCodesToStringTransformer;
+use Lakion\SyliusElasticSearchBundle\Search\Elastic\Factory\Query\QueryFactoryInterface;
+use ONGR\ElasticsearchDSL\Aggregation\FiltersAggregation;
+use ONGR\ElasticsearchDSL\Search;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Product\Model\ProductOptionValue;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class ProductOptionValueFilter extends AbstractType
+{
+    /**
+     * @var RepositoryManagerInterface
+     */
+    private $repositoryManager;
+
+    /**
+     * @var QueryFactoryInterface
+     */
+    private $productHasOptionCodeQueryFactory;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $productOptionValueRepository;
+
+    /**
+     * @param RepositoryManagerInterface $repositoryManager
+     * @param QueryFactoryInterface $productHasOptionCodeQueryFactory
+     * @param RepositoryInterface $productOptionValueRepository
+     */
+    public function __construct(
+        RepositoryManagerInterface $repositoryManager,
+        QueryFactoryInterface $productHasOptionCodeQueryFactory,
+        RepositoryInterface $productOptionValueRepository
+    ) {
+        $this->repositoryManager = $repositoryManager;
+        $this->productHasOptionCodeQueryFactory = $productHasOptionCodeQueryFactory;
+        $this->productOptionValueRepository = $productOptionValueRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(
+            'product_option_code',
+            EntityType::class, [
+                'class' => ProductOptionValue::class,
```

Comment:
This class should not be hardcoded, if someone customizes it, search will stop working. We should use the parameter here.